FIx:  Delete wc_layered_nav_counts_* transients whenever a product is updated

rodrigoprimo opened the pull request:
### Changes proposed in this Pull Request:

This PR moves the code to delete wc_layered_nav_counts_* transients from WC_Product_Data_Store_CPT::update_attributes() to wc_delete_product_transients(). This way those transients will be deleted whenever a product is updated and not only when the product attributes are updated. This is needed to make sure that the "Filter Products by Attribute" displays the correct list of attributes when a product changes from out of stock to in stock or vice versa via the "Quick Edit" interface or the WC API.

This PR also includes PHPCS fixes for all the modified files.

Closes #21429.

### How to test the changes in this Pull Request:

1. See #21429 for instructions on how to test this problem when updating products using the "Quick Edit" interface.
2. To test using the WC API, follow the same steps described in the issue above, but instead of using the "Quick Edit" interface, use the WC API to update a product stock quantity to zero (or from zero to any number >= 1).

### Changelog entry

> Fix: update "Filter Products by Attribute" widget when product stock quantity changes via "Quick Edit" or WC API

rodrigoprimo committed:
Fix phpcs violations automatically using phpcbf

rodrigoprimo committed:
Update WPCS sniff name from WordPress.WP.PreparedSQL.NotPrepared to WordPress.DB.PreparedSQL.NotPrepared

The name of the sniff changed in WPCS 1.0

rodrigoprimo committed:
Delete wc_layered_nav_counts_* transients whenever a product is updated

This commit moves the code to delete wc_layered_nav_counts_* transients from WC_Product_Data_Store_CPT::update_attributes() to wc_delete_product_transients(). This way those transients will be deleted whenever a product is updated and not only when the product attributes are updated. This is needed to make sure that the "Filter Products by Attribute" displays the correct list of attributes when a product changes from out of stock to in stock or vice versa via the "Quick Edit" interface or the WC API.

claudiulodro commented:
Overall this change seems to solve the reported issue. It does introduce a new bug, though. If you delete an attribute from a product the counts do not get updated. Here is a gif example:
![newattsbug](https://user-images.githubusercontent.com/7317227/50695019-54c0cd00-0ff0-11e9-87a8-02df5fd2022b.gif)

I think we should be able to keep the old transient clearing in `update_attributes` and this new one you've added. That should cover all cases.

claudiulodro reviewed (changes requested):
See comment about new bug.

rodrigoprimo committed:
Restore code to delete wc_layered_nav_counts_* transients

Commit a9994feeeac77d796fc57d955d004b4fd486b735 removed the code to delete wc_layered_nav_counts_* transients whenever a product attribute is updated and moved it to wc_delete_product_transients(). This was done to make sure those transients are deleted whenever a product is updated and not only when a product attribute changes. But, as Claudiu pointed out (https://github.com/woocommerce/woocommerce/pull/22029#issuecomment-451473512), this change introduced a bug as if you remove an attribute from a product, by the time the code reaches wc_delete_product_transients() the changes have been applied already to the product, the attribute doesn't belong to it anymore, and thus the layered nav counts transient for that attribute is not deleted.

To work around this problem, this commit restores the code that deletes wc_layered_nav_counts_* transients inside WC_Product_Data_Store_CPT::update_attributes(). Now, in most cases, WC will try to delete the same transient twice, but I can't see any other way to implement a solution for the problem above without doing this.

rodrigoprimo commented:
Good catch, I didn't notice that my change had broken the deletion of the transient when a product attribute is removed. I agree with you that restoring the code to remove transient inside WC_Product_Data_Store_CPT::update_attributes() while keeping the new code in wc_delete_product_transients() is the best approach. I just push a commit that implements this.

claudiulodro reviewed (approved):
Latest version looks good and works well in all situations. 👍
